pytest_plugins/auto_vault.py

```diff
@@ -0,0 +1,19 @@
+"""Plugin enables pytest to notify and update the requirements"""
+import os
+import re
+import subprocess
+from pathlib import Path
+
+
+def pytest_addoption(parser):
+    """Options to allow user to update the requirements"""
+    envpath = Path('.env')
+    # check if this is being executed by a user and not automation/CI
+    if (
+        re.search(r'\s*#.*VAULT_SECRET_ID_FOR_DYNACONF', envpath.read_text())
+        and 'VAULT_SECRET_ID_FOR_DYNACONF' not in os.environ
```

Comment:
I think this wouldn't handle the case where this variables are already set in dotenv?

Reply:
@Gauravtalreja1 The condition is now changed to just verify if `VAULT_SECRET_ID_FOR_DYNACONF` environment var is set as thats the feasible check as per our CI, Local execution and PR checks!